dns/src/lib.rs

```diff
@@ -466,18 +497,44 @@ impl Client {
 
     /// A simple DNS query.
     ///
-    /// This will only send a DNS query, it will not wait for a reply from the
-    /// DNS server.
+    /// This will only send a DNS or MDNS query, it will not wait for a reply.
     fn query<'a, W5500: Udp>(
         &mut self,
         w5500: &'a mut W5500,
     ) -> Result<Query<'a, W5500>, Error<W5500::Error>> {
-        w5500.udp_bind(self.sn, self.port)?;
+        let is_mdns = self.server == MDNS_SERVER;
+        if is_mdns {
```

Comment:
I am thinking `mdns` should be a separate `Client` structure in a `mdns` module, that uses the same underlying functions as the DNS client.  Then the `mdns::Client` struct also wont have the unused `rng` member. 
I can help out with this if you want.

Thoughts?

Reply:
No preference.